### ISSUE TITLE
Plugins: Bumping SystemJS to 6.14.2 broke image rendering

### ISSUE THREAD

**Issue by simPod** [grafana/grafana#76180]
### What happened?

I'm using image renderer plugin and was trying to render a panel from simpod-json-datasource.

I'm seeing these issues:

```
logger=context userId=1 orgId=1 uname=admin t=2023-10-09T09:40:53.551999241Z level=info msg="Request Completed" method=GET path=/grafana/public/plugins/simpod-json-datasource/module.js status=404 remote_addr=[::1] time_ms=7 duration=7.407245ms size=49009 referer="http://localhost:3000/d-solo/f62e9b8ce669da4dbd37ffb05efb0949/asnrender?from=1696502356000&height=400&kiosk=tv&orgId=1&panelId=1&render=1&to=1696509556000&tz=Europe%2FPrague&var-asn_raw=8075&width=900" handler=notfound
logger=context userId=1 orgId=1 uname=admin t=2023-10-09T09:40:53.551999241Z level=info msg="Request Completed" method=GET path=/grafana/public/plugins/simpod-json-datasource/module.js status=404 remote_addr=[::1] time_ms=7 duration=7.407245ms size=49009 referer="http://localhost:3000/d-solo/f62e9b8ce669da4dbd37ffb05efb0949/asnrender?from=1696502356000&height=400&kiosk=tv&orgId=1&panelId=1&render=1&to=1696509556000&tz=Europe%2FPrague&var-asn_raw=8075&width=900" handler=notfound
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.552985487Z level=error msg="Browser console error" msg="Failed to load resource: the server responded with a status of 404 (Not Found)" url="http://localhost:3000/grafana/public/plugins/simpod-json-datasource/module.js?_cache=0.6.2"
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.552985487Z level=error msg="Browser console error" msg="Failed to load resource: the server responded with a status of 404 (Not Found)" url="http://localhost:3000/grafana/public/plugins/simpod-json-datasource/module.js?_cache=0.6.2"
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.555495565Z level=error msg="Browser console error" msg="runRequest.catchError {\"message\":\"Datasource: 000000002 was not found\"}" url=http://localhost:3000/public/build/8721.e20fc3671a22764fd1f1.js line=4487 column=22241
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.555495565Z level=error msg="Browser console error" msg="runRequest.catchError {\"message\":\"Datasource: 000000002 was not found\"}" url=http://localhost:3000/public/build/8721.e20fc3671a22764fd1f1.js line=4487 column=22241
```

### What did you expect to happen?

Successful render

### Did this work before?

I managed to track the issue to this commit 62821c69b331b0982ac4de07f9da580ec7bbac18. Without it it works.

### How do we reproduce it?

1. build simpod-json-datasource from source
2. have a timeseries panel
3. try to render it as image using renderer plugin

### Environment (with versions)?

Grafana: 62821c69b331b0982ac4de07f9da580ec7bbac18
OS: linux debian
Browser: chromium?

### Grafana platform?

Other

### Datasource(s)?

simpod-json-datasource v0.6.2 (build from source)

**Comment by simPod** [comment]
Maybe something arond this line 🤔 https://github.com/grafana/grafana/commit/62821c69b331b0982ac4de07f9da580ec7bbac18#diff-5ec44332042eaf93388d58f7f569821340724239718e970d6ee413c366f8079fL46

**Comment by simPod** [comment]
cc @jackw

**Comment by simPod** [comment]
Just FYI, tried with latest grafana renderer plugin (3.8.something IIRC)
and also with 3.6.something. I don't think its version has any effect on
the issue. Which makes sense considering the nature of the commit with
systemjs bump.

**Comment by jackw** [comment]
Thanks for the extra info. Are you using a custom.ini at all? From the error message it looks like it's running on a subPath of `/grafana`?

```method=GET path=/grafana/public/plugins/simpod-json-datasource/module.js status=404```

**Comment by simPod** [comment]
Yes, I use

`root_url = https://%(domain)s/grafana`

**Comment by jackw** [comment]
I'm afraid I cannot reproduce this issue. I've tried:

- locally against latest `main` without a subpath using latest published versions of image renderer and simpod-json-datasource
- locally against latest `main` with a subpath using latest published versions of image renderer and simpod-json-datasource
- locally against latest `main` with and without a subpath against built from source simpod-json-datasource
- a hosted Grafana cloud instance running `Grafana v10.2.0-61719 (165b4c3f66)`

> Maybe something arond this line 🤔 [62821c6#diff-5ec44332042eaf93388d58f7f569821340724239718e970d6ee413c366f8079fL46](https://github.com/grafana/grafana/commit/62821c69b331b0982ac4de07f9da580ec7bbac18#diff-5ec44332042eaf93388d58f7f569821340724239718e970d6ee413c366f8079fL46)

This line of code is used by the plugin sandbox which is feature flagged. I don't think it's involved in this issue. 

I'm not all that familiar with the image renderer plugin but the paths in the logs look odd to me. Some reference the subpath `/grafana` but others (like the referrers) don't?

Do you have any custom setup for the image renderer or any other custom settings for Grafana?

**Comment by simPod** [comment]
I see. I'll try to create reproducible example. Not sure if I can manage
though since the renderer plugin does not work on arm arch.

**Comment by mckn** [comment]
@simPod I'm closing this for now. I'm happy to reopen when you have supplied some more details on how we can reproduce this on. OK?

**Comment by simPod** [comment]
Ok. I'm planing to try it with rendering service first and to ditch the renderer plugin. Will get back if there are still some issues.

**Comment by simPod** [comment]
@mckn please reopen 👇🏾 

@jackw did you try it with `serve_from_sub_path` set to `true`? Because it is reproducible only with `false`.

When I have `serve_from_sub_path` set to `false` and do the rewrite via nginx as specified in docs https://grafana.com/tutorials/run-grafana-behind-a-proxy/ `rewrite  ^/grafana/(.*)  /$1 break;` then you can reproduce the issue.

It works with `serve_from_sub_path: true` and no rewrite in proxy.

It does not matter whether you have rendering plugin installed or using a rendering service.

**Comment by jackw** [comment]
Thanks for taking the time to look into this further and updating us.

> did you try it with serve_from_sub_path set to true? Because it is reproducible only with false.

I hadn't attempted to repro it via a proxy. I'd only tried by using the Grafana http server. Having invested quite a bit of time tinkering with Grafana config and Nginx config try as we might we're really struggling to reproduce the exact errors that you mention:

```
logger=context userId=1 orgId=1 uname=admin t=2023-10-09T09:40:53.551999241Z level=info msg="Request Completed" method=GET path=/grafana/public/plugins/simpod-json-datasource/module.js status=404 remote_addr=[::1] time_ms=7 duration=7.407245ms size=49009 referer="http://localhost:3000/d-solo/f62e9b8ce669da4dbd37ffb05efb0949/asnrender?from=1696502356000&height=400&kiosk=tv&orgId=1&panelId=1&render=1&to=1696509556000&tz=Europe%2FPrague&var-asn_raw=8075&width=900" handler=notfound
logger=context userId=1 orgId=1 uname=admin t=2023-10-09T09:40:53.551999241Z level=info msg="Request Completed" method=GET path=/grafana/public/plugins/simpod-json-datasource/module.js status=404 remote_addr=[::1] time_ms=7 duration=7.407245ms size=49009 referer="http://localhost:3000/d-solo/f62e9b8ce669da4dbd37ffb05efb0949/asnrender?from=1696502356000&height=400&kiosk=tv&orgId=1&panelId=1&render=1&to=1696509556000&tz=Europe%2FPrague&var-asn_raw=8075&width=900" handler=notfound
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.552985487Z level=error msg="Browser console error" msg="Failed to load resource: the server responded with a status of 404 (Not Found)" url="http://localhost:3000/grafana/public/plugins/simpod-json-datasource/module.js?_cache=0.6.2"
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.552985487Z level=error msg="Browser console error" msg="Failed to load resource: the server responded with a status of 404 (Not Found)" url="http://localhost:3000/grafana/public/plugins/simpod-json-datasource/module.js?_cache=0.6.2"
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.555495565Z level=error msg="Browser console error" msg="runRequest.catchError {\"message\":\"Datasource: 000000002 was not found\"}" url=http://localhost:3000/public/build/8721.e20fc3671a22764fd1f1.js line=4487 column=22241
logger=plugin.grafana-image-renderer t=2023-10-09T09:40:53.555495565Z level=error msg="Browser console error" msg="runRequest.catchError {\"message\":\"Datasource: 000000002 was not found\"}" url=http://localhost:3000/public/build/8721.e20fc3671a22764fd1f1.js line=4487 column=22241
```
@simPod is there any chance you could provide your Grafana config and proxy config that would help us reproduce this more easily?

**Comment by simPod** [comment]
This was my nginx config

```
upstream grafana {
  server grafana:3000;
}

server {
    listen 80;

    error_log  /var/log/nginx/error.log;
    access_log /var/log/nginx/access.log;

    location /grafana/ {
        rewrite  ^/grafana/(.*)  /$1 break;
        proxy_set_header Host $http_host;
        proxy_pass http://grafana;
    }
```

and this grafana one

```
      GF_AUTH_ANONYMOUS_ENABLED: "true"
      GF_AUTH_ANONYMOUS_ORG_ROLE: Admin
      GF_AUTH_DISABLE_LOGIN_FORM: "true"
      GF_DATABASE_HOST: postgres
      GF_DATABASE_NAME: grafana
      GF_DATABASE_PASSWORD: postgres
      GF_DATABASE_TYPE: postgres
      GF_DATABASE_USER: postgres
      GF_LIVE_ALLOWED_ORIGINS: https://*b.c.com
      GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS: simpod-json-datasource
      GF_RENDERING_SERVER_URL: http://grafana-renderer:8081/render
      GF_RENDERING_CALLBACK_URL: http://grafana:3000/
      GF_SECURITY_ALLOW_EMBEDDING: "true"
      GF_SERVER_ROOT_URL: https://${slug}.a.b.c.com/grafana/
```

--------

I have migrated my setup to use `GF_SERVER_SERVE_FROM_SUB_PATH: "true"` and no rewrite on my proxy so I no longer suffer from the issue.

**Comment by mohan-rh07** [comment]
I see the fix is committed. I validated in latest 10.3.3, but the issue still exists. Can you please confirm the fix is available in 10.3.3?

**Comment by wbrowne** [comment]
> I see the fix is committed. I validated in latest 10.3.3, but the issue still exists. Can you please confirm the fix is available in 10.3.3?

Hi there, this fix is scheduled for `10.2.5` (yet to be scheduled), `10.3.4` (yet to be scheduled), `10.4.0` (scheduled for early March).

**Comment by mohan-rh07** [comment]
> > I see the fix is committed. I validated in latest 10.3.3, but the issue still exists. Can you please confirm the fix is available in 10.3.3?
> 
> Hi there, this fix is scheduled for `10.2.5` (yet to be scheduled), `10.3.4` (yet to be scheduled), `10.4.0` (scheduled for early March).

Thank you! Much appreciated.
